pandas/core/generic.py

```diff
@@ -4640,6 +4640,90 @@ def _reindex_with_indexers(
 
         return self._constructor(new_data).__finalize__(self)
 
+    def select_str(
+        self, *, startswith=None, endswith=None, regex=None, flags=0, axis=None
+    ):
+        """
+        Selects rows or columns of dataframe according to string labels in
+        the specified index.
+
+        Only one of keywords arguments `startswith`, `endswith` and `regex` can be used.
+
+        Parameters
+        ----------
+        startswith: str, optional
```

Comment:
Can we not just boil the signature down to `regex`?

Reply:
Yeah, startswith and endswith are not cardinal points, just thought they are often used, so nicer for people who don't want/know how to use regexes.